internal/provisioner/kops_provisioner.go

```diff
@@ -605,6 +570,128 @@ func makeClusterInstallationName(clusterInstallation *model.ClusterInstallation)
 	return fmt.Sprintf("mm-%s", clusterInstallation.Namespace[0:4])
 }
 
+// waitForHelmRunning is used to check when Helm is ready to install charts.
+func waitForNamespacesDeleted(ctx context.Context, namespaces []string, k8sClient *k8s.KubeClient) error {
+	for {
+		select {
+		case <-ctx.Done():
+			return errors.Wrap(ctx.Err(), "timed out waiting for namespaces to become fully terminated")
+		default:
+			var shouldWait bool
+			for _, namespace := range namespaces {
+				_, err := k8sClient.Clientset.CoreV1().Namespaces().Get(namespace, metav1.GetOptions{})
+				if err != nil && k8sErrors.IsNotFound(err) {
+					continue
+				}
+
+				shouldWait = true
+				break
+			}
+
+			if !shouldWait {
+				return nil
+			}
+
+			time.Sleep(5 * time.Second)
+		}
+	}
+}
+
+// waitForHelmRunning is used to check when Helm is ready to install charts.
+func waitForHelmRunning(ctx context.Context, configPath string) error {
+	for {
+		cmd := exec.Command("helm", "ls", "--kubeconfig", configPath)
```

Comment:
This applies in a bunch of places, but these exec functions do have versions that accept contexts in case this gets stuck.

Reply:
Yeah, the exec functions are a less-then-ideal stopgap we put in to get helm up and running. We plan to remove helm and all of this code later. Also, the helm/exec code isn't actually new in this PR, I just moved some functions to a better spot.

We will definitely improve this in the future.